server/cherry_pick.go

```diff
@@ -316,15 +328,65 @@ func (s *Server) addAssignee(ctx context.Context, newPRNumber int, pr *model.Pul
 	}
 }
 
-func returnToMaster(dir string) {
-	cmd := exec.Command("git", "checkout", "master")
+func returnToMaster(ctx context.Context, dir string) error {
+	cmd := exec.CommandContext(ctx, "git", "checkout", "master")
+	if err := runCommand(cmd, dir); err != nil {
+		return err
+	}
+	return nil
+}
+
+func cloneRepo(ctx context.Context, dir, upstreamSlug, originSlug, repoName string) error {
+	// Clone repo
+	cmd := exec.CommandContext(ctx, "git", "clone", "--depth=1", "git@github.com:"+originSlug+".git")
```

Comment:
I think `--depth=1` also implies `--single-branch`, so isn't only the master branch pulled? Doesn't this cause a failure when we check out the release branch in the cherry-pick.sh script?

Even if `--single-branch` isn't implied (older versions of git), then we still wont be able to checkout branches that are deeper than the given revisions old

The cherry-pick.sh script does a `git remote update`, but that doesn't fetch other branches or older commits.  This means that the local checkout might not have the commit we want to cherry-pick, if that commit is sufficiently old. 

I'm not a git expert, so my comments above need to be double checked.

Reply:
Yeah, it means only master branch. But when I ran the script, it downloaded some more stuff by itself and was able to successfully push a branch, just failing in the hub command. I didn't check how it did that though. Nevertheless, I'll remove `--depth=1` to be safe.